python/pyarrow/ipc.pxi

```diff
@@ -471,17 +471,22 @@ cdef class _CRecordBatchWriter(_Weakrefable):
         else:
             raise ValueError(type(table_or_batch))
 
-    def write_batch(self, RecordBatch batch):
+    def write_batch(self, RecordBatch batch, custom_metadata=None):
         """
         Write RecordBatch to stream.
 
         Parameters
         ----------
         batch : RecordBatch
+        custom_metadata : dict
```

Comment:
Should this be perhaps
```suggestion
        custom_metadata : KeyValueMetadata or dict
```

Reply:
Fixed

Reply:
I change it to `KeyValueMetadata` like other parts.